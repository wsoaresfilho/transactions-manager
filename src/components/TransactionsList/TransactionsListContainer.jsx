import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import TransactionsList from './TransactionsList';
import {
    deleteTransaction as deleteTransactionAction,
    fetchTransactionsData,
} from '../../actions';

export class TransactionsListContainer extends PureComponent {
    componentDidMount() {
        const { allTransactions, getAllTransactions } = this.props;
        if (allTransactions.length === 0) {
            getAllTransactions();
        }
    }

    render() {
        const { allTransactions, deleteTransaction, className } = this.props;
        return (
            <TransactionsList
                allTransactions={allTransactions}
                deleteTransaction={deleteTransaction}
                className={className}
            />
        );
    }
}

TransactionsListContainer.defaultProps = {
    className: null,
};

TransactionsListContainer.propTypes = {
    allTransactions: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.arrayOf(
            PropTypes.shape({
                description: PropTypes.string,
                value: PropTypes.number,
                id: PropTypes.string,
            })
        ),
    ]).isRequired,
    getAllTransactions: PropTypes.func.isRequired,
    deleteTransaction: PropTypes.func.isRequired,
    className: PropTypes.string,
};

const mapStateToProps = ({ transactions }) => {
    return {
        allTransactions: transactions.allTransactions,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        getAllTransactions: () => dispatch(fetchTransactionsData()),
        deleteTransaction: transactionId =>
            dispatch(deleteTransactionAction(transactionId)),
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TransactionsListContainer);
