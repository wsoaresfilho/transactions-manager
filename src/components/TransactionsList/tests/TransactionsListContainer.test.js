import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import TransactionsListContainer from '../TransactionsListContainer';

describe('RegisterTransactionContainer component', () => {
    let wrapper;
    let store;
    const initialState = {
        transactions: {
            allTransactions: [],
        },
    };
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(
            <Provider store={store}>
                <TransactionsListContainer />
            </Provider>
        );
    });

    describe('should render', () => {
        it('TransactionsListContainer component', () => {
            expect(wrapper.find('TransactionsList').length).toBe(1);
            expect(wrapper.length).toEqual(1);
        });

        it('snapshot correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
