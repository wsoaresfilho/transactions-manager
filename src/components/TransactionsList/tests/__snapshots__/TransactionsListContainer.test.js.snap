// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterTransactionContainer component should render snapshot correctly 1`] = `ReactWrapper {}`;
